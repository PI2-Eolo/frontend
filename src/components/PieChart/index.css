.pie-chart-container {
    display: flex;
    height: fit-content;
    width: fit-content;
    flex-direction: column;
    align-items: flex-start;
}

.pie-chart-content {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    height: fit-content;
    width: fit-content;
    margin-top: 1em;
}

.pie-chart-label {
    font-family: "Poppins Thin";
    font-weight: bold;
    font-size: 1.5em;
}

.pie-chart-label-list {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
}

.pie-chart-item-label-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    margin-top: 0.5em;
    margin-bottom: 0.5em;
}

.pie-char-label-color-bar {
    width: 1.8em;
    height: 0.4em;
    border-radius: 1em;
    background-color: #033d68;
    margin-right: 1em;
}

.pie-chart-value-container {
    display: flex;
    flex-direction: column;
    width: fit-content;
    align-items: flex-start;
}

.pie-chart-item-label {
    font-family: "Poppins Thin";
    font-weight: 100;
    font-size: 1em;
    white-space: nowrap;
}

.pie-chart-item-value {
    font-family: "Poppins Thin";
    font-weight: bold;
    font-size: 0.8em;
    white-space: nowrap;
}