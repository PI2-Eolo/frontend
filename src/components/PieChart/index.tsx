import "./index.css";
import { PieChart as GraphPieChart } from "react-minimal-pie-chart";

interface DataDetails {
  label: string;
  value: number;
  color: string;
}
interface PieChartProps {
  label?: string;
  unit?: string;
  data?: DataDetails[];
}

const PieChart = ({ label = "", data = [], unit = "" }: PieChartProps) => {
  return (
    <div className="pie-chart-container">
      <h2 className="pie-chart-label">{label}</h2>
      <div className="pie-chart-content">
        <GraphPieChart
          data={data.map((item) => ({
            title: item.label,
            value: item.value,
            color: item.color,
          }))}
          lineWidth={20}
          radius={30}
          rounded={true}
          paddingAngle={15}
          viewBoxSize={[80, 80]}
          center={[35, 30]}
        />
        <div className="pie-chart-label-list">
          {data.map((item) => (
            <div className="pie-chart-item-label-container">
              <span
                className="pie-char-label-color-bar"
                style={{ backgroundColor: item.color }}
              />
              <div className="pie-chart-value-container">
                <span className="pie-chart-item-label">{item.label}</span>
                <span className="pie-chart-item-value">
                  {item.value} {unit}
                </span>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default PieChart;
