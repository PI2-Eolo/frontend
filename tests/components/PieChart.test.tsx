import React from "react";
import renderer from "react-test-renderer";
import PieChart from "../../src/components/PieChart";

const mockData = [
  { label: "Kitchen", value: 241, color: "#7459D9" },
  { label: "Living room", value: 132, color: "#B9ABEB" },
  { label: "Other", value: 32, color: "#E3DEF7" },
];
describe("PieChart component test", () => {
  it("component snapshot", () => {
    const componentWithoutProps = renderer.create(<PieChart />);
    expect(componentWithoutProps.toJSON()).toMatchSnapshot();
    const componentWithProps = renderer.create(
      <PieChart data={mockData} label="Consumption details" unit="kW/h" />
    );
    expect(componentWithProps.toJSON()).toMatchSnapshot();
  });
});
