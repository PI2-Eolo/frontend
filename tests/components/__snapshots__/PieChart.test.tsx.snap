// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart component test component snapshot 1`] = `
<div
  className="pie-chart-container"
>
  <h2
    className="pie-chart-label"
  />
  <div
    className="pie-chart-content"
  >
    <svg
      height="100%"
      viewBox="0 0 80 80"
      width="100%"
    />
    <div
      className="pie-chart-label-list"
    />
  </div>
</div>
`;

exports[`PieChart component test component snapshot 2`] = `
<div
  className="pie-chart-container"
>
  <h2
    className="pie-chart-label"
  >
    Consumption details
  </h2>
  <div
    className="pie-chart-content"
  >
    <svg
      height="100%"
      viewBox="0 0 80 80"
      width="100%"
    >
      <path
        d="M 62 30 A 27 27 0 1 1 8.227584159527407 26.50175042845536"
        fill="none"
        stroke="#7459D9"
        strokeLinecap="round"
        strokeWidth={6}
        style={Object {}}
      >
        <title>
          Kitchen
        </title>
      </path>
      <path
        d="M 10.04524572117446 19.691739289121646 A 27 27 0 0 1 50.529425335116976 7.912968765335911"
        fill="none"
        stroke="#B9ABEB"
        strokeLinecap="round"
        strokeWidth={6}
        style={Object {}}
      >
        <title>
          Living room
        </title>
      </path>
      <path
        d="M 55.716817331922556 12.684877140610496 A 27 27 0 0 1 61.079997309804845 23.011885782231943"
        fill="none"
        stroke="#E3DEF7"
        strokeLinecap="round"
        strokeWidth={6}
        style={Object {}}
      >
        <title>
          Other
        </title>
      </path>
    </svg>
    <div
      className="pie-chart-label-list"
    >
      <div
        className="pie-chart-item-label-container"
      >
        <span
          className="pie-char-label-color-bar"
          style={
            Object {
              "backgroundColor": "#7459D9",
            }
          }
        />
        <div
          className="pie-chart-value-container"
        >
          <span
            className="pie-chart-item-label"
          >
            Kitchen
          </span>
          <span
            className="pie-chart-item-value"
          >
            241
             
            kW/h
          </span>
        </div>
      </div>
      <div
        className="pie-chart-item-label-container"
      >
        <span
          className="pie-char-label-color-bar"
          style={
            Object {
              "backgroundColor": "#B9ABEB",
            }
          }
        />
        <div
          className="pie-chart-value-container"
        >
          <span
            className="pie-chart-item-label"
          >
            Living room
          </span>
          <span
            className="pie-chart-item-value"
          >
            132
             
            kW/h
          </span>
        </div>
      </div>
      <div
        className="pie-chart-item-label-container"
      >
        <span
          className="pie-char-label-color-bar"
          style={
            Object {
              "backgroundColor": "#E3DEF7",
            }
          }
        />
        <div
          className="pie-chart-value-container"
        >
          <span
            className="pie-chart-item-label"
          >
            Other
          </span>
          <span
            className="pie-chart-item-value"
          >
            32
             
            kW/h
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
